FreeRTOS/FreeRTOS-Plus-TCP
# Include ICMP while calculating IP header checksum

Description
-----------
When `ipconfigDRIVER_INCLUDED_RX_IP_CHECKSUM == 0`:
While calculating the IP header checksum, the ICMP protocol was ignored. This means that an ICMP packet with incorrect CRC will be accepted by the TCP stack.

This PR fixes the above bug. Thanks, @evpopov for bringing this to our attention.

Test Steps
-----------


Related Issue
-----------
https://forums.freertos.org/t/freertos-tcp-receives-icmp-packets-with-bad-ip-checksum/11821


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
